HLTrigger/btau/plugins/HLTSumJetTagWithMatching.cc

```diff
@@ -0,0 +1,152 @@
+  event.getByToken(m_JetsToken, h_Jets);
+  if (saveTags())
+    filterproduct.addCollectionTag(m_Jets);
+
+  edm::Handle<JetTagCollection> h_JetTags;
+  event.getByToken(m_JetTagsToken, h_JetTags);
+
+  // check if the product this one depends on is available
+  auto const& handle = h_JetTags;
+  auto const& dependent = handle->keyProduct();
+  if (not dependent.isNull() and not dependent.hasCache()) {
+    // only an empty AssociationVector can have a invalid dependent collection
+    edm::Provenance const& dependent_provenance = event.getProvenance(dependent.id());
+    if (dependent_provenance.branchDescription().dropped())
+      // FIXME the error message should be made prettier
+      throw edm::Exception(edm::errors::ProductNotFound)
+          << "Product " << handle.provenance()->branchName() << " requires product "
+          << dependent_provenance.branchName() << ", which has been dropped";
+  }
+
+  //// Loop on jet tags and store values, sorting them in decreasing order
+  std::vector<float> jetTag_values;
+  for (auto const& jet : *h_JetTags)
+    jetTag_values.push_back(jet.second);
+  std::sort(jetTag_values.begin(), jetTag_values.end(), std::greater<>());
+
+  //// Select only good tags
+  std::vector<float> jetTag_values_selected;
+  unsigned int nJet = 0;
+  for (auto const& jetTag : jetTag_values) {
+    if (nJet >= m_MaxJetToSum)
+      break;
+    ++nJet;
+    jetTag_values_selected.push_back(jetTag);
+  }
+
+  /// produce the output jet collection
+  nJet = 0;
+  std::vector<TRef> jetRefCollection;
+  float sumJetTag = 0;
+  for (auto const& jet : *h_Jets) {
+    TRef jetRef = TRef(h_Jets, nJet);
+    auto tagValue = findTag(jet, *h_JetTags, m_deltaR);  // find the tag associated to the jet
+    LogTrace("") << "Jet " << nJet << " : Et = " << jet.et() << " , tag value = " << tagValue;
+    nJet++;
+    if (std::find(jetTag_values_selected.begin(), jetTag_values_selected.end(), tagValue) !=
+        jetTag_values_selected.end()) {  // find the tag
+      jetRefCollection.push_back(jetRef);
+      sumJetTag += tagValue;
+    }
+  }
+
+  if (m_UseMeanValue)
+    sumJetTag /= jetTag_values_selected.size();
+
+  // Accept value of the filter
+  bool accept = true;
+  if (jetRefCollection.size() < m_MinJetToSum)
+    accept = false;
+  else {
+    if (sumJetTag >= m_MinTag and sumJetTag <= m_MaxTag) {
+      accept = true;
+      for (auto const& jetRef : jetRefCollection) {
+        filterproduct.addObject(m_TriggerType, jetRef);
+      }
+    } else
+      accept = false;
+  }
+
+  edm::LogInfo("") << " trigger accept ? = " << accept << " nTag/nJet = " << jetRefCollection.size() << "/" << nJet
+                   << std::endl;
+  return accept;
+}
+
+typedef HLTSumJetTagWithMatching<reco::CaloJet> HLTSumCaloJetTagWithMatching;
+typedef HLTSumJetTagWithMatching<reco::PFJet> HLTSumPFJetTagWithMatching;
+DEFINE_FWK_MODULE(HLTSumCaloJetTagWithMatching);
+DEFINE_FWK_MODULE(HLTSumPFJetTagWithMatching);
```

Comment:
Same as above regarding `plugins/modules.cc` and moving templated members to the header.